### ISSUE TITLE
PCM::direct_mmap_capture errors with ENOTTY on ioctl

### ISSUE THREAD

**Issue by tranzystorekk** [diwic/alsa-rs#118]
After updating to version 0.9.0 I get an ENOTTY error when attempting to call `PCM::direct_mmap_capture()`:

```
Error: ALSA function 'SNDRV_PCM_IOCTL_CHANNEL_INFO' failed with error 'Not a typewriter (25)'
```

The environment is:

Void Linux (vmware guest)
Linux 6.6.18
Vmware Workstation Player 17.5.0

**Comment by diwic** [comment]
Interesting. Are you sure the difference is between two different alsa-rs versions and not something else in the environment (e g, not all drivers/devices support mmap)?

**Comment by tranzystorekk** [comment]
I'm pretty sure, this reproduces when I update this project from 0.8.1 to 0.9.0: https://codeberg.org/tranzystorekk/spectroscope, here's where I call the API: https://codeberg.org/tranzystorekk/spectroscope/src/commit/376cfbf1151a3d77615b20822965fb728ea5ffa9/src/capture.rs#L32

For an overview, the program attempts to open an alsa loopback device to read audio for visualizing.

**Comment by diwic** [comment]
@Jake-Shadle Hi, it's probably a regression from your "nuking nix" PR, would you mind having a look at this bug? Thanks!

**Comment by Jake-Shadle** [comment]
Sorry about that, fixed in #119

**Comment by tranzystorekk** [comment]
Just tested with the PR commit and the issue is fixed, thank you @Jake-Shadle ❤️
